services/src/pro/datasets/postgres.rs

```diff
@@ -167,6 +178,7 @@ where
     }
 
     async fn load_dataset(&self, dataset: &DatasetId) -> Result<Dataset> {
+        self.update_dataset_status(dataset).await?;
```

Comment:
can you reuse the connection from the line below?

Reply:
2865e61058c4a931cf41c73e497f2e812bc78b17